android/src/main/kotlin/clean/news/ui/item/share/ItemShareIntent.kt

```diff
@@ -0,0 +1,48 @@
+package clean.news.ui.item.share
+
+import android.content.Intent
+import clean.news.core.entity.Item
+import clean.news.flow.WithComponent
+import clean.news.flow.WithIntent
+import clean.news.presentation.navigation.NavigationFactory.ItemShareKey
+import clean.news.ui.item.detail.ItemDetailScreen
+import clean.news.ui.item.detail.ItemDetailScreen.ItemDetailComponent
+import dagger.Module
+import dagger.Provides
+import dagger.Subcomponent
+import flow.ClassKey
+import flow.TreeKey
+import nz.bradcampbell.paperparcel.PaperParcel
+import nz.bradcampbell.paperparcel.PaperParcelable
+
+@PaperParcel
+class ItemShareIntent(val item: Item) : ClassKey(),
```

Comment:
Yeah, I guess you're right about the platform agnostic point. However, I think the `Intent` descriptor is more if an implementation detail and is addressed by the `WithIntent` implementation. I think these key implementations should be named by what they are (screens) not how they accomplish their presentation (view/intent/etc). Thoughts?